### DISCUSSION TITLE
Modal Action Without Cancel Button

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#5211] [Help] [answered]
Is it possible to create a modal action without a cancel button? Or without a submit button? I'm basically using this as a preview, and there really isn't any further action other than closing it. So ideally, I only want one button that says "Close".

My code:

```php
Pages\Actions\Action::make('preview_badge')
    ->label('Preview Badge')
    ->icon('heroicon-o-identification')
    ->color('secondary')
    ->action(function (array $data): void {
        // nothing
    })
    ->form([
        Forms\Components\ViewField::make('badge_preview')
            ->view('filament.forms.components.badge-preview'),
    ])
    ->modalHeading('Badge Preview')
    ->modalButton('Close')
    ->modalWidth('sm'),
```

**Comment by danharrin** [comment]
```php
Pages\Actions\ViewAction::make('preview_badge')
    ->label('Preview Badge')
    ->icon('heroicon-o-identification')
    ->form([
        Forms\Components\ViewField::make('badge_preview')
            ->view('filament.forms.components.badge-preview'),
    ])
    ->modalHeading('Badge Preview')
    ->modalWidth('sm'),
```

**Reply by danharrin** [reply]
You might also need to pass `->record($this->record)` if you want the record data in the form

**Reply by intrepidws** [reply]
This actually links directly to the View page for me. I assume that in order to do what you have suggested, I would need to remove my "view" page entirely? Unfortunately, that isn't something that I can/want to do here.

Would you be open to a PR to allow an action modal without a cancel button?

**Reply by danharrin** [reply]
No, just pass in `->url(null)` :)

**Reply by intrepidws** [reply]
Unfortunately, that doesn't seem to change anything for me. I do appreciate your help though.

**Reply by danharrin** [reply]
Try `->url()->action(fn () => null)`. I'm convinced it works 😆

**Reply by intrepidws** [reply]
That doesn't do what I need either - but maybe we're saying different things. I'm trying to hide either the Cancel or the Submit button. The code you provided makes it so that the Cancel and Submit buttons basically do the same thing - which is nothing.

What I ended up doing, which is admittedly a bit of a hack, is hiding the cancel button on this specific modal with a CSS rule:

```css
div.filament-page.filament-resources-create-record-page.filament-resources-registrants div.filament-modal-window div.filament-modal-footer button:first-of-type,
div.filament-page.filament-resources-edit-record-page.filament-resources-registrants div.filament-modal-window div.filament-modal-footer button:first-of-type {
    @apply hidden;
}
```

**Comment by intrepidws** [comment]
For anyone who stumbles on this discussion and is looking for a solution, this is what I ended up doing.

> What I ended up doing, which is admittedly a bit of a hack, is hiding the cancel button on this specific modal with a CSS rule:
> 
> ```css
> div.filament-page.filament-resources-create-record-page.filament-resources-registrants div.filament-modal-window div.filament-modal-footer button:first-of-type,
> div.filament-page.filament-resources-edit-record-page.filament-resources-registrants div.filament-modal-window div.filament-modal-footer button:first-of-type {
>     @apply hidden;
> }
> ```

**Reply by danharrin** [reply, marked as answer]
The alternative is `->modalActions(fn ($action) => [$action->getModalCancelAction()->label('Close')])`

**Reply by intrepidws** [reply]
OK that's much better - thanks Dan.
